src/renderer/components/conversations/formatRelativeTime.ts

```diff
@@ -1,59 +1,74 @@
-import moment from 'moment'
-import 'moment/min/locales'
+import { DateTime } from 'luxon'
 
-const getExtendedFormats = () => ({
-  y: 'lll',
-  M: `${window.static_translate('timestamp_format_m_desktop') || 'MMM D'} LT`,
-  d: 'ddd LT',
-})
-const getShortFormats = () => ({
-  y: 'll',
-  M: window.static_translate('timestamp_format_m_desktop') || 'MMM D',
-  d: 'ddd',
-})
-
-function isToday(timestamp: moment.Moment) {
-  const today = moment().format('ddd')
-  const targetDay = moment(timestamp).format('ddd')
-  return today === targetDay
-}
-
-function isYear(timestamp: moment.Moment) {
-  const year = moment().format('YYYY')
-  const targetYear = moment(timestamp).format('YYYY')
-  return year === targetYear
-}
+import type { getMessageFunction } from '../../../shared/localize'
 
 export default function formatRelativeTime(
   rawTimestamp: number,
   options: { extended: boolean }
 ) {
+  /* extended is how to display it in messages, so still rather short */
   const { extended } = options
   const tx = window.static_translate
-  const formats = extended ? getExtendedFormats() : getShortFormats()
-  const timestamp = moment(rawTimestamp)
-  const now = moment()
-  const diff = moment.duration(now.diff(timestamp))
+  const timestamp = DateTime.fromMillis(rawTimestamp)
+  const now = DateTime.now()
+  const diff = DateTime.now().diff(timestamp).shiftToAll()
 
-  if (diff.years() >= 1 || !isYear(timestamp)) {
-    return timestamp.format(formats.y)
-  } else if (diff.months() >= 1 || diff.days() > 6) {
-    return timestamp.format(formats.M)
-  } else if (diff.days() >= 1 || !isToday(timestamp)) {
-    return timestamp.format(formats.d)
-  } else if (diff.hours() >= 1) {
-    const key = 'n_hours'
+  if (diff.years >= 1 || now.year !== timestamp.year) {
+    return extended
+      ? timestamp.toLocaleString({
+          ...DateTime.DATETIME_SHORT,
+        }) /*10/14/1983, 1:30 PM*/
+      : timestamp.toLocaleString(DateTime.DATE_SHORT) /* 10/14/1983 */
+  } else if (diff.months >= 1 || diff.weeks >= 1 || diff.days >= 6) {
+    return extended
+      ? timestamp.toLocaleString({
+          ...DateTime.TIME_SIMPLE,
+          ...DateTime.DATE_FULL,
+          year: undefined,
+        }) /* June 1 at 5:56 PM */
+      : timestamp.toLocaleString({
+          ...DateTime.DATE_FULL,
+          year: undefined,
+          weekday: 'short',
+          month: 'short',
+        }) /* Jun 1 */
+  } else if (diff.days >= 1 || now.ordinal !== timestamp.ordinal) {
+    return extended
+      ? timestamp.toLocaleString({
+          ...DateTime.TIME_SIMPLE,
+          weekday: 'short',
+        }) /* Saturday, 5:51 PM */
+      : timestamp.toLocaleString({
+          ...DateTime.TIME_SIMPLE,
+          weekday: 'short',
+        }) /* Sat 5:51 PM */
+  } else {
+    timestamp.toRelative()
+  }
```

Comment:
But even with that, this format looks wonky, too verbose, takes too much space. This formats as "20 minutes ago". This is used both in the chat list and on messages, and it takes too much space IMO. It's the "ago" word I think.
I have checked out the docs of [`Intl.RelativeTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat) and of `timestamp.toRelative()` and have not found an option that gets rid of "ago". `{ style: 'narrow' }` formats it as "1m ago", and `{ style: 'short' }` as "1 min. ago".

Perhaps using just the time (13:37) is good, as most messengers (including DC Android) do. But this could be done in a separate MR I guess.